src/agreement/agreement-pool.ts

```diff
@@ -1,16 +1,16 @@
-import { Agreement } from "./agreement";
+import { Agreement, AgreementOptions } from "./agreement";
 import { createPool, Factory, Options as GenericPoolOptions, Pool } from "generic-pool";
 import { defaultLogger, Logger } from "../shared/utils";
 import { DraftOfferProposalPool } from "../market/draft-offer-proposal-pool";
-import { MarketModule, GolemMarketError, MarketErrorCode, MarketOptions } from "../market";
+import { MarketModule, GolemMarketError, MarketErrorCode } from "../market";
 import { AgreementDTO } from "./service";
 import { EventEmitter } from "eventemitter3";
-import { GolemUserError } from "../shared/error/golem-error";
+import { PaymentModule } from "../payment";
 
 export interface AgreementPoolOptions {
   logger?: Logger;
-  market: MarketOptions;
-  pool?: GenericPoolOptions;
+  poolOptions?: GenericPoolOptions;
```

Comment:
Just make sure that these "generic pool options" are not leaked out to the user-code, we don't want these to be the dependency of our user.

Reply:
it does not leak from the deployment API level, but if the user wants to use `AgreementPool` directly as in the `basic/hello-world.ts` example, he will be able to set custom parameters for the generic pool there..